components/message.php

```diff
@@ -0,0 +1,504 @@
+			'to'        => '',
+			'subject'   => '',
+			'content'   => '',
+			'thread-id' => false,
+			'date-sent' => bp_core_current_time(),
+			'silent'    => false,
+		) );
+
+		// Fill in the content.
+		if ( empty( $r['content'] ) ) {
+			$r['content'] = $this->generate_random_text();
+		}
+
+		// Fill in the subject.
+		if ( empty( $r['subject'] ) ) {
+			$r['subject'] = sprintf( 'Message Subject' );
+		}
+
+		$user = $this->get_user_id_from_identifier( $assoc_args['from'] );
+		if ( ! $user ) {
+			WP_CLI::error( 'No user found by that username or ID.' );
+		}
+
+		// To is not required when thread id is set.
+		if ( ! empty( $r['to'] ) ) {
+			$recipient = $this->get_user_id_from_identifier( (int) $r['to'] );
+
+			if ( ! $recipient ) {
+				WP_CLI::error( 'No user found by that username or ID.' );
+			}
+		}
+
+		// Existing thread recipients will be assumed.
+		$recipient = ( ! empty( $r['thread_id'] ) ) ? array() : array( $recipient->ID );
+
+		$thread_id = messages_new_message( array(
+			'sender_id'  => $user->ID,
+			'thread_id'  => $r['thread-id'],
+			'recipients' => $recipient,
+			'subject'    => $r['subject'],
+			'content'    => $r['content'],
+			'date_sent'  => $r['date-sent'],
+		) );
+
+		if ( ! is_numeric( $thread_id ) ) {
+			WP_CLI::error( 'Could not add a message.' );
+		}
+
+		if ( $r['silent'] ) {
+			return;
+		}
+
+		if ( \WP_CLI\Utils\get_flag_value( $assoc_args, 'porcelain' ) ) {
+			WP_CLI::line( $thread_id );
+		} else {
+			WP_CLI::success( 'Message successfully created.' );
+		}
+	}
+
+	/**
+	 * Delete message thread(s) for a given user.
+	 *
+	 * ## OPTIONS
+	 *
+	 * <thread-id>...
+	 * : Thread ID(s).
+	 *
+	 * --user-id=<user>
+	 * : Identifier for the user. Accepts either a user_login or a numeric ID.
+	 *
+	 * [--yes]
+	 * : Answer yes to the confirmation message.
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     $ wp bp message delete 500 687867 --user-id=40
+	 *     Success: Thread successfully deleted.
+	 *
+	 *     $ wp bp message delete 564 5465465 456456 --user-id=user_logon --yes
+	 *     Success: Thread successfully deleted.
+	 *
+	 * @alias remove
+	 */
+	public function delete( $args, $assoc_args ) {
+		$thread_id = $args[0];
+
+		$user = $this->get_user_id_from_identifier( $assoc_args['user-id'] );
+		if ( ! $user ) {
+			WP_CLI::error( 'No user found by that username or ID.' );
+		}
+
+		WP_CLI::confirm( 'Are you sure you want to delete this thread(s) ?', $assoc_args );
+
+		parent::_delete( array( $thread_id ), $assoc_args, function( $thread_id ) {
+
+			if ( messages_delete_thread( $thread_id, $user->ID ) ) {
+				return array( 'success', 'Thread successfully deleted.' );
+			} else {
+				return array( 'error', 'Could not delete the thread.' );
+			}
+		} );
+	}
+
+	/**
+	 * Get a message.
+	 *
+	 * ## OPTIONS
+	 *
+	 * <message-id>
+	 * : Identifier for the message.
+	 *
+	 * [--fields=<fields>]
+	 * : Limit the output to specific fields.
+	 * ---
+	 * default: All fields.
```

Comment:
When you use this syntax, wp-cli takes `All fields` as the literal value. I think we need to remove the `default` declaration here.